mlflow/pyfunc/__init__.py

```diff
@@ -265,7 +268,51 @@ def _load_model_env(path):
     return _get_flavor_configuration(model_path=path, flavor_name=FLAVOR_NAME).get(ENV, None)
 
 
-def load_model(model_uri, suppress_warnings=True):
+PyFuncOutput = Union[pandas.DataFrame, pandas.Series, np.ndarray, list]
+
+
+class PyFuncModel(object):
```

Comment:
Do we want this class to be public?

Reply:
Yes we do. I think it makes sense to make this the official pyfunc  interface. It has consequences though - e.g. some of our tests were passing in model_input=..., where model_input is only known to PythonModel. This will no longer work. We could consider passing in generic arguments instead to preserve such workloads, but I think it's better to enforce unified interface.